### ISSUE TITLE
Investigate accuracy discrepancy between evaluate_classifier method and model.evaluate

### ISSUE THREAD

**Issue by moosmiddelkoop** [SimonIlic/WeightSpaceClassifier#7]
## Summary

In `src/cnn_surgery/utils/evaluate_per_class_accuracy.py`, the `evaluate_classifier` function computes overall accuracy manually:

```python
logits = model.predict(x_test, batch_size=batch_size, verbose=0)
y_pred = np.argmax(logits, axis=1)
y_test_flat = y_test.flatten()
overall_acc = float(np.mean(y_pred == y_test_flat))
# NOTE: overall accuracy slightly differs from original model.evaluate method
```

This calculation may produce slightly different results compared to Keras's `model.evaluate()` method.

## Suggested action

- Investigate the source of the discrepancy
- Document whether the difference is acceptable for our use case
- Either align the implementations or document why the difference exists

## Location

`src/cnn_surgery/utils/evaluate_per_class_accuracy.py:61-62`
